src/ray/raylet/node_manager.cc

```diff
@@ -638,6 +651,7 @@ void NodeManager::ProcessGetTaskMessage(
 
 void NodeManager::ProcessDisconnectClientMessage(
     const std::shared_ptr<LocalClientConnection> &client) {
+  RAY_LOG(INFO) << "ProcessDisconnectClientMessage Started!";
   // Remove the dead worker from the pool and stop listening for messages.
   const std::shared_ptr<Worker> worker = worker_pool_.GetRegisteredWorker(client);
```

Comment:
Further investigation shows that `task_id.is_nil()` evaluates to true when going through error handling (non-intentional disconnecting)